analyses/ssgsea-hallmark/02-analyze-plot-ssgsea-res.R

```diff
@@ -0,0 +1,186 @@
+#Author: Pichai Raman
+#Date: 9/23/2019
+##########################################
+
+#Call libraries
+library("tidyverse")
+library("pheatmap")
+library("viridis")
+library("optparse")
+
+# Declare command line options
+option_list <- list(
+  optparse::make_option(
+    c("-a", "--anova_pvalue"),
+    type = "double",
+    default = 0.01,
+    help = "ANOVA P-value threshold",
+  ),
+  optparse::make_option(
+    c("-t", "--tukey_pvalue"),
+    type = "double",
+    default = 0.05,
+    help = "Tukey HSD Adjusted P-value threshold"
+  ),
+  optparse::make_option(
+    c("-p", "--perc_keep"),
+    type = "double",
+    default = .25,
+    help = "Percentage of diseases a particular disease has a significantly higher score than for a specific pathway"
+  )
+)
+
+#Read the arguments passed
+opt_parser <- optparse::OptionParser(option_list = option_list)
+opt <- optparse::parse_args(opt_parser)
+
+#Pass arguments to variables
+aovParam <- opt$anova_pvalue
+padjParam <- opt$tukey_pvalue
+percKeep <- opt$perc_keep
+
+
+#Read in clinical data and pathway level data generated from 01-run-ssgsea.R
+clinData <- read.delim("../../data/pbta-histologies.tsv", stringsAsFactors=F)
+geneSetExpMat <- readRDS("../../scratch/GeneSetExpressionMatrix.RDS")
+
+#Choose Clinical Samples corresponding to samples in Gene Set Matrix
+#This is to makes sure that clinical data and the pathway DF have the same samples
+#and that they are ordered the same way
+clinData <- clinData[clinData[,"experimental_strategy"]=="RNA-Seq",]
+rownames(clinData) <- clinData[,"Kids_First_Biospecimen_ID"]
+clinData <- clinData[colnames(geneSetExpMat),]
+
+#Function to split pathway names from TukeyHSD into vector of 2 elements
+#i.e. Tukey HSD outputs Pathway1-Pathway2 and this will spitout c(Pathway1, Pathway2)
+#Will be used so that we can tell quickly which pathways are consistently upregulated vs other pathways
+getVarNames <- function(x)
+{
+	tmp <- strsplit(x, split="-")[[1]]
+	tmp1 <- tmp[1]
+	tmp2 <- tmp[2]
+	return(c(tmp1, tmp2))
+}
+
+#Analyse Pathways statistically correlated with clinical variables
+#Returns ANOVA P-value and Tukey HSD Results
+#Requires Pathway (pathwayName) and Clinical Variable (clinVarName)
+compareClinVarToPathway <- function(pathwayName=NULL, clinVarName=NULL)
+{
+	tmpDat <- data.frame(clinData[,clinVarName], geneSetExpMat[pathwayName,])
+	colnames(tmpDat) <- c("ClinVar", "PathwayVar") 	#Need to set this since pathwayName and clinVarName will change which is not good for aov function
+	aovOut <- aov(PathwayVar ~ ClinVar, data=tmpDat)
+	posthoc <- TukeyHSD(x=aovOut, conf.level=0.95)
+	posthoc <- data.frame(posthoc[[1]])
+	aovOutNamed <- unlist(summary(aovOut))
+	posthoc[,"F_Value"] <- aovOutNamed["F value1"]
+	posthoc[,"AOV_P"] <- aovOutNamed["Pr(>F)1"]
+	posthoc[,"Pathway"] <- pathwayName
+	posthoc[,"ClinVar"] <- clinVarName
+	posthoc <- cbind(t(sapply(rownames(posthoc), FUN=getVarNames)), posthoc) #Pulls out disease names 
```

Comment:
Can you make this comment more general? `clinVarName` may not always relate to diseases.

Reply:
Sure, updated.